contracts/yellow/IVault.sol

```diff
@@ -0,0 +1,113 @@
+//SPDX-License-Identifier: Unlicense
+pragma solidity 0.8.16;
+
+/**
+ * @notice IVault is the interface to implement custody.
+ */
+interface IVault {
+    // ======================
+    // Structs
+    // ======================
+    struct Allocation {
+        address asset;
+        uint256 amount;
+    }
+
+    // ======================
+    // Functions
+    // ======================
+    /**
+     * @notice Get last ledger id (deposits and withdrawals id).
+     * @return uint256 Ledger id.
+     */
+    function getLastId() external view returns (uint256);
+
+    /**
+     * @notice Set the address derived from the broker's new public key. Emits `BrokerKeyDerivedAddressSet` event.
+     * @dev Supplied payload must be signed by broker's current public key.
+     * @param encodedAddress Encoded address to set.
+     * @param signature Encoded address signed by broker's current public key.
+     */
+    function setBrokerKeyDerivedAddress(bytes calldata encodedAddress, bytes calldata signature) external;
+
+    /**
+     * @notice Set the address derived from the OTP's new public key. Emits `OTPKeyDerivedAddressSet` event.
+     * @dev Supplied payload must be signed by OTP's current public key.
+     * @param encodedAddress Encoded address to set.
+     * @param signature Encoded address signed by OTP's current public key.
+     */
+    function setOTPKeyDerivedAddress(bytes calldata encodedAddress, bytes calldata signature) external;
+
+    /**
+     * @notice Deposit assets with given payload from the caller. Emits `Deposited` event.
+     * @param payload Encoded payload, which consists of rid (unique identifier id), expire timestamp, deposit address and an array of allocations.
+     * @param brokerSignature Payload signed by the Broker.
+     * @param otpSignature Payload signed by the OTP service.
+     * @return bool Return 'true' if deposited successfully.
+     */
+    function deposit(
+        bytes calldata payload,
```

Comment:
define a struct for the Payload, see the example in Duckies.sol

Reply:
68d0b0199c3213fd4f6726ee4543adc8b162d0a6